src/qml/LocatorItem.qml

```diff
@@ -159,81 +160,59 @@ Item {
     id: busyIndicator
     z: 11
     running: locator.isRunning
-    anchors.right: barcodeReaderButtonRect.left
+    anchors.right: barcodeReaderButton.left
     anchors.rightMargin: -15
-    anchors.verticalCenter: barcodeReaderButtonRect.verticalCenter
+    anchors.verticalCenter: barcodeReaderButton.verticalCenter
     height: searchFieldRect.height - 10
     visible: false
   }
 
-  Rectangle {
-    id: clearButtonRect
+  QfToolButton {
+    id: clearButton
+    anchors.centerIn: busyIndicator
+    visible: false
+
     width: 40
     height: 40
+    padding: 2
     z: 12
-    color: "transparent"
-    anchors.centerIn: busyIndicator
-    visible: false
 
-    Image {
-      id: clearButton
-      z: 12
-      width: 20
-      height: 20
-      source: Theme.getThemeIcon("ic_clear_black_18dp")
-      sourceSize.width: 20 * screen.devicePixelRatio
-      sourceSize.height: 20 * screen.devicePixelRatio
-      fillMode: Image.PreserveAspectFit
-      anchors.centerIn: clearButtonRect
-      opacity: searchField.displayText.length > 0 ? 1 : 0.25
-    }
+    iconSource: Theme.getThemeIcon("ic_clear_black_18dp")
+    iconColor: Theme.mainTextColor
+    bgcolor: "transparent"
+    opacity: searchField.displayText.length > 0 ? 1 : 0.25
 
-    MouseArea {
-      anchors.fill: parent
-      onClicked: {
-        if (searchField.displayText.length > 0) {
-          searchButton.forceActiveFocus();
-          searchField.text = '';
-          searchField.forceActiveFocus();
-        } else {
-          locatorItem.state = "off"
-        }
+    onClicked: {
+      if (searchField.displayText.length > 0) {
+        searchButton.forceActiveFocus();
+        searchField.text = '';
+        searchField.forceActiveFocus();
+      } else {
+        locatorItem.state = "off"
       }
     }
   }
 
-  Rectangle {
-    id: barcodeReaderButtonRect
+  QfToolButton {
+    id: barcodeReaderButton
     width: 40
     height: 40
+    padding: 2
     z: 12
-    color: "transparent"
     anchors.right: searchButton.left
     anchors.rightMargin: 5
     anchors.verticalCenter: searchFieldRect.verticalCenter
     visible: false
 
-    Image {
-      id: barcodeReaderButton
-      z: 12
-      width: 24
-      height: 24
-      source: Theme.getThemeVectorIcon("ic_qrcode_black_24dp")
-      sourceSize.width: 24 * screen.devicePixelRatio
-      sourceSize.height: 24 * screen.devicePixelRatio
-      fillMode: Image.PreserveAspectFit
-      anchors.centerIn: barcodeReaderButtonRect
-      opacity: 1
-    }
+    iconSource: Theme.getThemeVectorIcon("ic_qrcode_black_24dp")
+    iconColor: Theme.darkTheme ? Theme.mainTextColor : "transparent"
```

Comment:
hm, so we are hiding this in non-dark DarkTheme? A comment why will help.

Reply:
I'll add a note in QfToolButton, in this context, Qt handles "transparent" as a do-no-colorize keyword, and the original SVG color is used to render the icon. In this case, the original QR code icon had a nice green touch on top of black, which I'm preserving on light theme.